### PR TITLE
Add Java Spring Boot Guide

### PR DESCRIPTION
**Pull request by unicodeveloper**
Spring Boot Java Guide

CleanShot 2024-10-22 at 21 16 16@2x

### CODE REVIEW

File: src/docs/guides/spring-boot.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+Railway offers multiple ways to deploy your Spring Boot app, depending on your setup and preference. 
+
+### One-Click Deploy from a Template
+
+If you’re looking for the fastest way to get started, the one-click deploy option is ideal. 
+
+Click the button below to begin:
+
+[![Deploy on Railway](https://railway.app/button.svg)](https://railway.app/new/template/JvYvDw)
```
Comment (brody192) [inline src/docs/guides/spring-boot.md:99]: I personally dont think we should link to templates from creators we don't know or don't have any contact with.

maybe you could follow your guide and publish a spring boot template under your account?

File: src/docs/guides/spring-boot.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    FROM eclipse-temurin:17-jdk-alpine
+
+    # Create and change to the app directory.
+    WORKDIR /app
+
+    # Copy local code to the container image.
+    COPY . ./
+
+    # Build the app.
+    RUN chmod +x ./mvnw && ./mvnw -DoutputFile=target/mvn-dependency-list.log -B -DskipTests clean dependency:list install
```
Comment (brody192) [inline src/docs/guides/spring-boot.md:168]: do we need to chmod here? doesnt the image come with mvnw that is already an executable? if it doesn't, can we use a better image?

File: src/docs/guides/spring-boot.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    - Click [Generate Domain](/guides/public-networking#railway-provided-domain) to create a public URL for your app.
+
+### Use a Dockerfile
+
+1. Create a `Dockerfile` in the `helloworld` or Spring Boot app's root directory.
+2. Add the content below to the `Dockerfile`:
+    ```bash
+    # Use the Node alpine official image
+    # https://hub.docker.com/_/eclipse-temurin
+    FROM eclipse-temurin:17-jdk-alpine
```
Comment (brody192) [inline src/docs/guides/spring-boot.md:159]: isn't java 17 quite old by now?

Comment (unicodeveloper) [reply src/docs/guides/spring-boot.md:159]: Well, kinda but our nickpacks java provider still runs Java 17 by default.

Comment (brody192) [reply src/docs/guides/spring-boot.md:159]: yeah but dockerfiles are better than nixpacks ;)

File: src/docs/guides/spring-boot.md (hunk 2 above)
Comment (unicodeveloper) [reply src/docs/guides/spring-boot.md:168]: Good point. I followed the build command Nixpacks use by default to build Java apps. I'm gonna check this without the chmod.

File: src/docs/guides/spring-boot.md (hunk 1 above)
Comment (unicodeveloper) [reply src/docs/guides/spring-boot.md:99]: Oh, this guy is not active like Vin? I assumed he was.

File: src/docs/guides/spring-boot.md (hunk 3 above)
Comment (unicodeveloper) [reply src/docs/guides/spring-boot.md:159]: Because it's more flexible and offer more options? 🤔

Comment (brody192) [reply src/docs/guides/spring-boot.md:159]: yep!

File: src/docs/guides/spring-boot.md (hunk 1 above)
Comment (brody192) [reply src/docs/guides/spring-boot.md:99]: i dont know how active they are on the platform itself, or if they even check on their templates, but they are not active in the community

File: src/docs/guides/spring-boot.md (hunk 3 above)
Comment (unicodeveloper) [reply src/docs/guides/spring-boot.md]: Upgraded to 21.

File: src/docs/guides/spring-boot.md (hunk 1 above)
Comment (unicodeveloper) [reply src/docs/guides/spring-boot.md]: Switched with the new template I published.

**Pull request merged**